pkg/kubelet/eviction/eviction_manager.go

```diff
@@ -145,6 +150,41 @@ func (m *managerImpl) IsUnderInodePressure() bool {
 	return hasNodeCondition(m.nodeConditions, api.NodeInodePressure)
 }
 
+func startMemoryThresholdNotifier(thresholds []Threshold, observations signalObservations, hard bool, handler thresholdNotifierHandlerFunc) error {
+	for _, threshold := range thresholds {
+		if threshold.Signal != SignalMemoryAvailable ||
+			(hard && threshold.GracePeriod != time.Duration(0)) ||
+			(!hard && threshold.GracePeriod == time.Duration(0)) {
+			continue
+		}
+		observed, found := observations[SignalMemoryAvailable]
+		if !found {
+			continue
+		}
+		cgroups, err := cm.GetCgroupSubsystems()
+		if err != nil {
+			return err
+		}
+		// TODO add support for eviction from --cgroup-root
+		cgpath, found := cgroups.MountPoints["memory"]
```

Comment:
nit: I'd prefer moving all cgroups logic to under `/pkg/kubelet/cm` to have better abstractions. I'm OK to perform that refactor in a separate PR though given that v1.5 FF is this week.